src/formatDate.js

```diff
@@ -1,14 +1,34 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      receivedDate['YY'] = dateArr[i].slice(2);
+    }
+
+    if (fromFormat[i] === 'YY') {
+      receivedDate['YYYY'] = (dateArr[i] < 30 ? 20 : 19) + dateArr[i];
+    }
+
+    receivedDate[fromFormat[i]] = dateArr[i];
```

Comment:
This line will overwrite the 'YYYY' or 'YY' value in the `receivedDate` object even if it was correctly set in the previous conditionals. You should add an else clause to prevent this from happening.